var/spack/repos/builtin/packages/py-pycairo/package.py

```diff
@@ -11,14 +11,16 @@ class PyPycairo(PythonPackage):
     """Pycairo is a set of Python bindings for the cairo graphics library."""
 
     homepage = "https://www.cairographics.org/pycairo/"
-    url      = "https://github.com/pygobject/pycairo/releases/download/v1.17.1/pycairo-1.17.1.tar.gz"
-    url      = "https://files.pythonhosted.org/packages/68/76/340ff847897296b2c8174dfa5a5ec3406e3ed783a2abac918cf326abad86/pycairo-1.17.1.tar.gz"
+    url      = "https://pypi.io/packages/source/p/pycairo/pycairo-1.17.1.tar.gz"
 
+    version('1.18.1', sha256='70172e58b6bad7572a3518c26729b074acdde15e6fee6cbab6d3528ad552b786')
     version('1.17.1', sha256='0f0a35ec923d87bc495f6753b1e540fd046d95db56a35250c44089fbce03b698')
 
-    depends_on('cairo@1.2.0:')
+    depends_on('cairo@1.13.1:')
     depends_on('pkgconfig', type='build')
     depends_on('py-setuptools', type='build')
+    depends_on('python@2.7:2.8,3.3:', when=':1.17.1',  type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.4:', when='@1.18.1:', type=('build', 'run'))
```

Comment:
I would remove these Python version constraints, as they aren't reinforced anywhere. If you do want to keep them anyway, you're missing an `@` symbol.